mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -2764,7 +2782,12 @@ static void processMapMembersWithParent(
         mapData.DevicePointers[memberDataIdx]);
     combinedInfo.Names.emplace_back(
         LLVM::createMappingInformation(memberClause.getLoc(), ompBuilder));
-    combinedInfo.BasePointers.emplace_back(mapData.BasePointers[mapDataIndex]);
+    if (checkIfPointerMap(memberClause))
+      combinedInfo.BasePointers.emplace_back(
+          mapData.BasePointers[memberDataIdx]);
+    else
+      combinedInfo.BasePointers.emplace_back(
+          mapData.BasePointers[mapDataIndex]);
```

Comment:
Nit: Consider refactoring this a bit, since at the moment it's a bit difficult to tell at first glance what the difference is between the two branches. Feel free to ignore if you disagree.
```suggestion
    uint64_t basePointerIndex = checkIfPointerMap(memberClause)? memberDataIdx : mapDataIndex;
    combinedInfo.BasePointers.emplace_back(mapData.BasePointers[basePointerIndex]);
```